pkg/arvo/sys/zuse.hoon

```diff
@@ -6036,6 +6037,57 @@
       ?~  myn  ~
       ?:  =(~ t.myn)  (trip i.myn)
       (weld (trip i.myn) `tape`['/' $(myn t.myn)])
+    ::
+    ::  |base16: en/decode arbitrary MSB-first hex strings
+    ::
+    ++  base16
+      ~%  %base16  +  ~
+      |%
+      ++  en
+        ~/  %en
+        |=  a=(pair @ud @)  ::  $octs, but safe
+        ^-  cord
+        ::  trailing zero bytes, possible truncated input
+        ::
+        =/  [pad=@ud dat=@]
+          =/  wid  (met 3 q.a)
+          ?:  (gte p.a wid)
+            [(sub p.a wid) q.a]
+          =/  dat  (cut 3 [0 p.a] q.a)
+          =.  wid  (met 3 dat)
+          [(sub p.a wid) dat]
```

Comment:
I think this should eventually be factored out into a standard helper, so we can apply this pattern consistently whenever we're rendering `$octs`.